src/transformers/models/whisper/modeling_tf_whisper.py

```diff
@@ -1386,3 +1387,69 @@ def prepare_inputs_for_generation(
             "decoder_attention_mask": decoder_attention_mask,
             "decoder_position_ids": decoder_position_ids,
         }
+
+
+class TFWhisperForAudioClassification(TFWhisperPreTrainedModel):
```

Comment:
This LGTM from a functionality point of view! Will leave it to one of the TF maintainers to assist you with the TF-specific semantics, but what you've got here looks complete to me (no need for the freeze encoder method since this is only for fine-tuning, which we don't have in TF)

Reply:
Ok, great! I still see that there are some tests failing from within `TFModelTesterMixin`, so I can try to resolve those.